atoma-state/src/handlers.rs

```diff
@@ -819,30 +819,28 @@ pub(crate) async fn handle_update_stack_num_compute_units_and_claim_funds(
         event = "handle-update-stack-num-compute-units-and-claim-funds",
         "Processing update stack num compute units and claim funds"
     );
-    let (
-        UpdateStackNumComputeUnitsAndClaimFunds {
-            ratio,
-            stack_computed_units,
-            is_confidential,
-            is_locked_for_claim,
-        },
-        concurrent_requests,
-    ) = {
-        let count = { *concurrent_requests.entry(stack_small_id).or_insert(0) };
-        (
-            state_manager
-                .state
-                .update_stack_num_compute_units(
-                    stack_small_id,
-                    estimated_total_compute_units,
-                    total_compute_units,
-                    RATIO_FOR_CLAIM_STACK_THRESHOLD,
-                    count as i64,
-                )
-                .await?,
-            count,
-        )
+    let entry = concurrent_requests.entry(stack_small_id);
+    let concurrent_requests = match &entry {
+        Entry::Occupied(entry) => *entry.get(),
+        Entry::Vacant(_entry) => 0, // If it was zero it was deleted, so vacant is treated as zero, don't do insert, because there is no cleanup
     };
+
+    let UpdateStackNumComputeUnitsAndClaimFunds {
+        ratio,
+        stack_computed_units,
+        is_confidential,
+        is_locked_for_claim,
+    } = state_manager
+        .state
+        .update_stack_num_compute_units(
+            stack_small_id,
+            estimated_total_compute_units,
+            total_compute_units,
+            RATIO_FOR_CLAIM_STACK_THRESHOLD,
+            concurrent_requests as i64,
+        )
+        .await?;
```

Comment:
For the sake of debugging could we add

```rust
{
    Ok(result) => result,
    Err(err) => {
        // Log detailed information about the error context
        tracing::error!(
            "Failed to update stack compute units for stack_id={}: error={:?}, concurrent_requests={}",
            stack_small_id,
            err,
            request_count
        );
        drop(entry); // Drop the lock before propagating the error
        return Err(err);
    }
};
```

around the database operation to make debugging easier if something fails.

Reply:
Done